tensordict/tensordict.py

```diff
@@ -3005,7 +3005,14 @@ def _set(self, key: str, value, inplace: bool = False) -> TensorDictBase:
         else:
             td, subkey = self, key
         if inplace:
-            td.get(subkey).copy_(value)
+            try:
+                td.get(subkey).copy_(value)
+            except KeyError as err:
+                raise err
+            except Exception as err:
```

Comment:
Can we be more specific than this? Or is the problem that we don't know what error the module will raise if passed a `None` value where something concrete is expected?

Reply:
we still raise the error, we just say which key is culprit (to help debugging)
Why would we need to be more specific?

Reply:
Yeah ok, I think this is actually ok. I was a bit wary because `except Exception` will catch everything, but as you say because we reraise it perhaps there's no danger here.